src/SqlPersistence/CommandWrapper.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+class CommandWrapper : IDisposable
+{
+    protected DbCommand command;
+
+    public CommandWrapper(DbCommand command)
+    {
+        this.command = command;
+    }
+
+    public DbCommand InnerCommand => command;
+
+    public string CommandText
+    {
+        get { return command.CommandText; }
+        set { command.CommandText = value; }
+    }
+
+    public DbTransaction Transaction
+    {
+        get { return command.Transaction; }
+        set { command.Transaction = value; }
+    }
+
+    public virtual void AddParameter(string name, object value)
+    {
+        var parameter = command.CreateParameter();
```

Comment:
doesnt making this an extension method on `DbCommand` render this class redundant?